### PR TITLE
2.2 student registration pitsyk

### PR DESCRIPTION
**Pull request by staspitsyk**
1. Created endpoint for students registration
2. Created validation for students data fields
3. Created seeders for students, teachers, groups tables for testing endpoints

### CODE REVIEW

File: src/modules/teachers/teachers.model.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ export class Teachers extends Model {
   public isAdmin: boolean;
   public createdAt: Date;
   public updatedAt: Date;
+  static init: any;
```
Comment (Alan18081) [inline src/modules/teachers/teachers.model.ts:15]: Why do we need this property

File: src/modules/students/students.service.ts [hunk 2]
```diff
@@ -1,13 +1,66 @@
...
+    }
+
+    const group = await groupsService.findOneById(groupId);
+
+    if (!group) {
+      throw new NotFound("Group not found");
+    }
+
+    if (group.groupToken !== groupToken) {
+      throw new BadRequest("Not walid group token");
```
Comment (Alan18081) [inline src/modules/students/students.service.ts:42]: You have misspelled word `valid`

File: src/modules/students/students.service.ts [hunk 3]
```diff
@@ -1,13 +1,66 @@
...
+      throw new BadRequest("User with provided email already exists");
+    }
+
+    const isExistSudentEmail = await this.findOneByEmail(email);
+
+    if (isExistSudentEmail) {
+      throw new BadRequest("User with provided email already exists");
+    }
+
+    const group = await groupsService.findOneById(groupId);
```
Comment (Alan18081) [inline src/modules/students/students.service.ts:35]: Search by `groupToken`, not by `groupId`

File: src/config/sequelize.config.ts [hunk 4]
```diff
@@ -11,7 +11,7 @@ module.exports = {
   },
   test: {
     username: process.env.DB_USER,
-    password: process.env.DB_PASS,
+    assword: process.env.DB_PASS,
```
[the remark this answers is not in the document]
Comment (staspitsyk) [reply src/config/sequelize.config.ts]: Fixed typo

File: src/common/exeptions/base-http-error.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import { ValidationError } from 'joi';
+import { ValidationError } from "joi";
```
[the remark this answers is not in the document]
Comment (staspitsyk) [reply src/common/exeptions/base-http-error.ts]: Done

File: src/common/passport/auth.strategy.ts [hunk 6]
```diff
@@ -1,29 +1,29 @@
...
+      done(null, false);
     }
+  } catch (err) {
+    done(new Unauthorized(err.message), false);
+  }
 });
 
-//passport.use(strategy);
\ No newline at end of file
+//passport.use(strategy);
```
[the remark this answers is not in the document]
Comment (staspitsyk) [reply src/common/passport/auth.strategy.ts]: Removed

File: src/modules/teachers/teachers.service.ts [hunk 7]
```diff
@@ -1,13 +1,21 @@
...
 
 class TeachersService {
-  async findOneByEmail(email: string) {
+  async findOneByEmail(email) {
     const teacher = await Teachers.findOne({
-      where: { email }
+      where: { email },
     });
-    
+    console.log(teacher);
```
[the remark this answers is not in the document]
Comment (staspitsyk) [reply src/modules/teachers/teachers.service.ts]: Done

File: src/modules/teachers/teachers.model.ts (hunk 1 above)
Comment (staspitsyk) [reply src/modules/teachers/teachers.model.ts]: Removed

File: src/modules/students/students.service.ts (hunk 2 above)
Comment (staspitsyk) [reply src/modules/students/students.service.ts]: Fixed typo

File: src/modules/students/students.service.ts (hunk 3 above)
Comment (staspitsyk) [reply src/modules/students/students.service.ts]: Changed group search by groupToken

**Pull request merged**